Feather_Freezer_Alarm/code.py

```diff
@@ -0,0 +1,263 @@
+            #if both doors are closed
+            if not rightswitch.value and not leftswitch.value:
+                print('doors just closed')
+                #if prelarm is true then set it to False
+                if prealarm is True:
+                    buzzer1.value = False
+                #if an alarm is true then upload to IO alarm resolved
+                if alarm1 or alarm2:
+                    #publish 0 to alarm feed
+                    io.publish(alarmfeed, 0)
+                    #buzzers off/Alarms to False
+                    buzzer1.value = False
+                    buzzer2.value = False
+                    alarm1 = False
+                    alarm2 = False
+                    #toggle alarm resolved feed to send email notification
+                    io.publish(resolvedfeed, 1)
+                    time.sleep(5)
+                    io.publish(resolvedfeed, 0)
+
+        #check motion sensor if there is no alarm
+        if(alarm1 is False and alarm2 is False and prealarm is False):
+            #update pir sensor
+            motion.update()
+            #if motion stopped
+            if motion.fell:
+                print('motion stopped')
+                #publish 0 to motion feed
+                io.publish(motionfeed, 1)
+                time.sleep(.25)
+                io.publish(motionfeed, 0)
+            #if motion started
+            if motion.rose:
+                print('motion detected')
+                #reset start times
+                start1 = time.monotonic()
+                start2 = time.monotonic()
+            #if continued motion
+            elif motion.value:
+                print('still motion')
+                io.publish(motionfeed, 1)
+                time.sleep(5)
+            #if continued no motion
+            else:
+                print('no motion')
+
+        print("\n")
+
+        # Explicitly pump the message loop.
+        io.loop()
+
+        #check difference between time now and start times if more than N seconds start beeping
+        if (((time.monotonic() - start1) >= 300) or ((time.monotonic() - start2) >= 300)):
+            prealarm = True
+            #beeping
+            buzzer1.value = True
+            time.sleep(.5)
+            buzzer1.value = False
+
+        #check if difference between time now and start1 if more than X seconds turn on buzzer2
+        if (alarm1 is False and ((time.monotonic() - start1) >= 600)):
+            alarm1 = True
+            buzzer2.value = True
+            #publish 1 to alarm feed
+            io.publish(alarmfeed, 1)
+
+        #check if difference between time now and start2 if more than X seconds turn on buzzer2
+        if (alarm2 is False and ((time.monotonic() - start2) >= 600)):
+            alarm2 = True
+            buzzer2.value = True
+            #publish 1 to alarm feed
+            io.publish(alarmfeed, 1)
+
+        print(str(time.time()))
+        print(str(int(time.time()/300)))
```

Comment:
Are these debugging printouts that could be removed?
```suggestion
```

Reply:
@dhalbert - I think I now have these changed, redoing CI

Reply:
yes they can be removed